src/SourceBridgeClient.ts

```diff
@@ -10,28 +11,49 @@ export class SourceBridgeClient {
   // Map where we keep track of event subscriptions
   private messageCallbacks: Map<string, OnMessageFn[]> = new Map()
 
-  constructor() {
-    window.addEventListener('message', (event) => {
-      // Because we allow async callbacks, handleEnvelope is async, but addEventListener only takes non-async callbacks
-      // so we use `void` operator to allow using an async function here.
-      void this.handleEnvelope(event)
-    })
+  constructor(private readonly otherWindow: Window) {
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    window.addEventListener('message', (message) => this.handleMessage(message))
+    // Because we allow async callbacks, handleEnvelope is async, but addEventListener only takes non-async callbacks
+    // so we use `void` operator to allow using an async function here.
+  }
+
+  public close(): void {
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    window.removeEventListener('message', this.handleMessage)
   }
 
-  public sendEvent(message: EventEnvelope): void {
-    console.log('[SourceBridge] sending message to parent: ', message)
-    parent.postMessage(JSON.stringify(message), '*')
+  // Because we allow async callbacks, handleEnvelope is async, but addEventListener only takes non-async callbacks
+  // so we use `void` operator to allow using an async function here.
+  private handleMessage(event: MessageEvent<unknown>): void {
+    void this.handleEnvelope(event)
   }
 
-  public async sendRequest<TResponse>(message: EventEnvelope): Promise<TResponse> {
+  public sendEvent(message: Envelope): void {
+    console.log('[SourceBridge] sending message: ', message)
```

Comment:
do we want these console logs to stay?

Reply:
They can be useful for debugging but generally customers won't want them. I will refactor into a `debug()` method that is disabled by default.